crates/aptos-crypto/src/noise.rs

```diff
@@ -666,7 +655,7 @@ impl NoiseSession {
             .ok_or(NoiseError::NonceOverflow)?;
 
         // return a subslice without the authentication tag
-        Ok(authentication_tag.to_vec())
+        Ok(authentication_tag.as_ref().try_into().unwrap())
```

Comment:
what is the type for the authentication_tag here that may fail the conversion?

Reply:
It's `ring` type called `Tag`:

```rust
#[must_use]
#[repr(C)]
pub struct Tag([u8; TAG_LEN]);

impl AsRef<[u8]> for Tag {
    fn as_ref(&self) -> &[u8] {
        self.0.as_ref()
    }
}
```

I think this is just me not being a good enough Rustacean. The following works also:

```rust
        Ok(authentication_tag.as_ref().into())
```

Reply:
you can use to_vec() directly

Reply:
I cannot:

```
error[E0599]: no method named `to_vec` found for struct `ring::aead::Tag` in the current scope
   --> crates/aptos-crypto/src/noise.rs:654:31
    |
654 |         Ok(authentication_tag.to_vec())
    |                               ^^^^^^ method not found in `ring::aead::Tag`

```